src/DGtal/geometry/curves/AlphaThickSegmentComputer.h

```diff
@@ -0,0 +1,700 @@
+   * alpha_max. If it is the case the new point is added and the
+   * segment parameters are updated, otherwise the alpha thick segment
+   * is keep in its original state.
+   *
+   * @return 'true' if the segment has been extended and 'false'
+   * otherwise (the object is then in its original state).
+   */
+  bool extendFront();
+
+
+  /**
+   *  Tests whether the current alpha thick segment can be extended at
+   *  the front with the next contour point i.e checks if we have
+   *  still an alpha thick segment of width alpha_max after adding the
+   *  given point \aPoint. The segment parameters are keep in its
+   *  original state.
+   *  
+   * @return 'true' if the segment can be extended with the given point, 'false' otherwise.
+   */
+  bool isExtendableFront();
+
+  
+  /**
+   *  Tests whether the current alpha thick segment can be extended, i.e
+   *  checks if we have still an alpha thick segment of width alpha_max after
+   *  adding the given point \aPoint. The segment parameters are keep
+   *  in its original state.
+   *  
+   * @param[in] aPoint the point to be tested for the segment extension.
+   * @return 'true' if the segment can be extended with the given point, 'false' otherwise.
+   */
+  bool isExtendableFront(const InputPoint &aPoint);
+
+
+  /**
+   * Tries to add the point \a aPoint at the front of the current alpha thick
+   * segment and checks if we have still an alpha thick segment of thickness
+   * less or equal to the initial value alpha_max. If it is the case
+   * the new point is added and the segment parameters are updated,
+   * otherwise the alpha thick segment is keep in its original state.
+   *
+   * @param[in] aPoint the new point to extend the current segment.
+   * @return 'true' if the segment has been extended and 'false'
+   * otherwise (the object is then in its original state).
+   */
+  bool extendFront(const InputPoint &aPoint);
+  
+ 
+
+  //-------------------- Primitive services -----------------------------
+  
+  
+  /**
+     Returns the current primitive recognized by this computer,
+     which is a ParallelStrip of axis width smaller than the one
+     specified at instanciation.
+  */
+  Primitive primitive() const;
+  
+  
+
+  
+    // ----------------------- Interface --------------------------------------
+public:
+  
+  
+  /**
+   * Checks the validity/consistency of the object.
+   * @return 'true' if the object is valid, 'false' otherwise.
+   */
+  bool isValid() const;
+  
+
+  /**
+   * Computes the paralell strip params from the current state of the segment.
+   * @param[out] mu the minimal value of N.X (with N is the normal vector of the segment).
+   * @param[out] N the normal of the vector (not normalized).
+   * @param[out] nu the width of the strip. 
+   *
+   **/
+  void computeParalellStripParams(double &mu, PointD &N, double &nu) const;
+  
+ 
+   /**
+   * @return the segment length defined from the basic bouding box (@see getBasicBoundingBox).
```

Comment:
well, I propose to rename:
-  getBasicBoundingBox -> getBoundingBoxFromExtremPoints (same name as the one with point parameters)
- getRealBoundingBox -> getBoundingBox
Is it ok for you  ?